portal-api/src/v1/gef/controllers/application-submit.js

```diff
@@ -34,18 +34,32 @@ const generateId = async ({
   user,
 });
 
-const generateUkefDealId = async (application) => application.ukefDealId || generateId({
-  entityId: application._id,
-  entityType: 'deal',
-  dealId: application._id,
-  user: application.checkerId,
-});
+const generateUkefDealId = async (application) => {
+  if (application.ukefDealId) {
```

Comment:
I think this condition (same for `generateUkefFacilityId` function) should be outside of this because otherwise it's not actually generating one. I'd vote to do it like this, what do you think?

```js
// application-submit.js
const addSubmissionData = async (dealId, existingApplication) =>  {
  // ...
  await addSubmissionDateToIssuedFacilities(dealId);
  // ...

  const submissionData = {
    submissionCount: count,
    submissionDate: date,
    // ...
  };

  if (!existingApplication.ukefId) {
    const { ukefId } = await generateUkefDealId(existingApplication);
    submissionData.ukefId = ukefId;
  }

  return submissionData;
};

```

Reply:
This would mean I would also need to add an else for returning `existingApplication.ukefId`?

Reply:
I don't think so - after this function is called, the GEF mongo query does a "find one and update" so it will only add/update and fields you pass to it. So if the ukefId already exists and you don't pass it in, it won't get deleted it'll just remain the same